server.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client.js');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const ee = new EE();
+
+const pool = [];
+
+ee.on('@dm', function(client, string) {
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').slice(1).join(' ').trim();
+
+  pool.forEach( c => {
+
+
+    if (c.nickname === nickname) {
+      c.socket.write(`${client.nickname}: ${message}`);
+    }
+  });
+});
+
+ee.on('@all', function(client, string) {
+  pool.forEach( c => {
+      c.socket.write(`${client.nickname}:` + string);
+  });
+});
+
+ee.on('default', function(client, string) {
+  client.socket.write('not a command\n');
+});
+
+server.on('connection', function(socket) {
+  var client = new Client(socket);
+  pool.push(client);
+
+
+  socket.on('data', function(data) {
+    const command = data.toString().split(' ').shift().trim();
+
+    // console.log(check);
+    // console.log(command);
+    // const message = data.toString().split(' ');
+    // console.log(message);
+    if (command.startsWith('@')) {
+      ee.emit(command, client, data.toString().split().slice(1).join(' '));

```

Comment:
@GLSea1979 the reason your code is currently not functioning is you are missing a separator argument in your `.split` method. So when your conditional runs it doesn't know where to split the data it gets back.

if you update your code to be `.split(' ')` where the separator is a space then your code will work.